README.md

```diff
@@ -0,0 +1,124 @@
+# OAuth2 Proxy K8s Operator
+
+This is the Kubernetes Python Operator for the
+[OAuth2 proxy](https://oauth2-proxy.github.io/oauth2-proxy/).
+
+## Description
+
+OAuth2 Proxy is a reverse proxy and static file server that authenticates users
+through providers like Google and GitHub, allowing validation by email, domain,
+or group.
+
+This operator provides the OAuth2 proxy, and consists of Python scripts which
+wraps the versions distributed by
+[OAuth2 proxy](https://quay.io/repository/oauth2-proxy/oauth2-proxy?tab=tags&tag=latest).
+
+## Usage
+
+The OAuth2 Proxy charm can be used to enable authentication for charmed and
+non-charmed applications, by providing the oauth2 configuration to the charm,
+and setting the `upstream` config value to the name of your application's k8s
+service. For charmed applications, this is the name of the deployed application.
+
+### Enable TLS
+
+To enable TLS connections, you must have a TLS certificate stored as a k8s
+secret (default name is oauth2-proxy-tls”). The secret name can be configured
+using the `tls-secret-name` config property in the charm. A self-signed
+certificate for development purposes can be created as follows:
+
+```bash
+# Generate private key
+openssl genrsa -out server.key 2048
+
+# Generate a certificate signing request
+openssl req -new -key server.key -out server.csr -subj "/CN=oauth2-proxy-k8s"
+
+# Create self-signed certificate
+openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt -extfile <(printf "subjectAltName=DNS:oauth2-proxy-k8s")
+
+# Create a k8s secret
+kubectl -n <model-name> create secret tls oauth2-proxy-tls --cert=server.crt --key=server.key
+```
+
+### Deploy
+
+To deploy Charmed OAuth2 Proxy, you need to run the following commands, which
+will enable ingress in your microk8s, fetch the charm from
+[Charmhub](https://charmhub.io/nginx-ingress-integrator) and deploy it to your
+model:
+
+```bash
+# Deploy ingress controller.
+sudo microk8s enable ingress:default-ssl-certificate=<model-name>/oauth2-proxy-tls
+
+# Deploy charms
+juju deploy oauth2-proxy-k8s --channel edge
+juju deploy nginx-ingress-integrator --channel edge --revision 103 --trust
+
+# Set the necessary config
```

Comment:
Maybe it could be useful to have some information of how to configure google OAuth 2.0 or a link to some documentation on it. I mean google as it is the default provider (there is a link in the charmcraft.yaml file)